lib/Sema/ConstraintSystem.h

```diff
@@ -2875,9 +2881,14 @@ class ConstraintSystem {
       if (formBindingScore(y) < formBindingScore(x))
         return false;
 
-      // If the only difference is default types,
+      // If there is a difference in number of default types,
       // prioritize bindings with fewer of them.
-      return x.NumDefaultableBindings < y.NumDefaultableBindings;
+      if (x.NumDefaultableBindings != y.NumDefaultableBindings)
+        return x.NumDefaultableBindings < y.NumDefaultableBindings;
+
+      // As a last resort, let's check if the bindings are
+      // potentially incomplete, and if so, let's de-prioritize them.
+      return x.PotentiallyIncomplete < y.PotentiallyIncomplete;
```

Comment:
Is there a reason that this cannot just be part of the binding score?

Reply:
Unfortunately yes, it has to be checked after NumDefaultableBindings is not a part of the score at the moment...

I can look into adding both to the score separately

Reply:
Ah, I see. Okay then I think this is fine for now and if you want to follow-up with that later that's okay.